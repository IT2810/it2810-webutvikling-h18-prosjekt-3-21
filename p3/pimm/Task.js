import React from "react";
import {
  Alert,
  AppRegistry,
  AsyncStorage,
  Platform,
  StyleSheet,
  Text,
  TouchableHighlight,
  TouchableOpacity,
  TouchableNativeFeedback,
  TouchableWithoutFeedback,
  View,
  Image
} from "react-native";

export class Task extends React.Component {
  constructor(props) {
    super(props);
    this.state.taskdescription = props.taskdescription;
  }

  state = {
    taskdescription: "Hent melk",
    pressed: false
  };

  componentDidMount() {
    _storeData = async () => {
      try {
        await AsyncStorage.setItem("Task1", "Hallo der du!");
      } catch (error) {
        Alert.alert("Error saving data!");
      }
    };

    _retrieveData = async () => {
      try {
        const value = await AsyncStorage.getItem("Task1");
        if (value != null) {
          console.log(value);
        }
      } catch (error) {
        Alert.alert("Error retrieving data");
      }
    };

    _storeData();
    _retrieveData();
  }

  _onPressCompleteTask = () => {
    Alert.alert("Complete task");
  };

  _onPressDeleteTask = () => {
    Alert.alert("Delete task");
  };

  _onPressEditTask = () => {
    Alert.alert("Edit task");
  };

  _toggleTaskOptions = () => {
    let s = !this.state.pressed;
    this.setState({ pressed: s });
  };

  render() {
    let taskItem;
    if (this.state.pressed === false) {
      taskItem = (
        <TouchableOpacity onPress={this._toggleTaskOptions} style={styles.task}>
          <Text style={styles.taskdescription}>
            {this.state.taskdescription}
          </Text>
        </TouchableOpacity>
      );
    } else {
      taskItem = (
        <View style={styles.taskoptions}>
          <TouchableOpacity onPress={this._toggleTaskOptions}>
            <Image
              source={require("./assets/back.png")}
              style={styles.taskoptionbutton}
            />
          </TouchableOpacity>

          <TouchableOpacity onPress={this._onPressEditTask}>
            <Image
              source={require("./assets/edit.png")}
              style={styles.taskoptionbutton}
            />
          </TouchableOpacity>

          <TouchableOpacity onPress={this._onPressDeleteTask}>
            <Image
              source={require("./assets/delete.png")}
              style={styles.taskoptionbutton}
            />
          </TouchableOpacity>

          <TouchableOpacity onPress={this._onPressCompleteTask}>
            <Image
              source={require("./assets/accept.png")}
              style={styles.taskoptionbutton}
            />
          </TouchableOpacity>
        </View>
      );
    }
    return taskItem;
  }
}

const styles = StyleSheet.create({
  task: {
    flexDirection: "row",
    backgroundColor: "#eee",
    alignItems: "center",
    justifyContent: "center",
    width: "100%",
    height: 100,
    borderColor: "#999",
    borderWidth: 3,
    borderRadius: 5,
    paddingLeft: 10,
    paddingRight: 10
  },

  taskoptions: {
    flexDirection: "row",
    backgroundColor: "#eee",
    alignItems: "center",
    justifyContent: "space-between",
    width: "100%",
    height: 100,
    borderColor: "#999",
    borderWidth: 3,
    borderRadius: 5,
    paddingLeft: 10,
    paddingRight: 10
  },

  taskoptionbutton: {
    width: 48,
    height: 48
  },

  taskdescription: {
    color: "#333",
    flex: 5
  }
});
