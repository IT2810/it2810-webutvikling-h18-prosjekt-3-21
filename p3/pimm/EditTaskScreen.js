import React, { Component } from "react";
import {
  Text,
  View,
  TouchableOpacity,
  StyleSheet,
  TextInput,
  Picker
} from "react-native";
import { styles } from "./HomeScreen";

export class EditTaskScreen extends Component {
  constructor(props) {
    super(props);
    this.state.parent = props.parent;
    this.state.taskDescription;
    this.state.taskType = "regTask";
  }
  state = {};
  render() {
    return (
      <View style={styles.container}>
        <View style={styles.header}>
          <TouchableOpacity style={styles.button}>
            <View>
              <Text style={styles.buttonText}>Add task</Text>
            </View>
          </TouchableOpacity>
        </View>
        <View style={styles.mainContainer}>
          <TextInput
            style={stylesEditScreen.textInputField}
            placeholder="Add task description here"
            blurOnSubmit={true}
            multiline={true}
            placeholderTextColor={"gray"}
            onChangeText={text => this.setState({ taskDescriptions: text })}
          />
          <View style={stylesEditScreen.pickerField}>
            <Text>Pick task type:</Text>
            <Picker
              selectedValue={this.state.taskType}
              onValueChange={(itemValue, itemIndex) =>
                this.setState({ taskType: itemValue })
              }
            >
              <Picker.Item label="Regular" value="regTask" />
              <Picker.Item label="Steps" value="stepTask" />
            </Picker>
          </View>
        </View>
        <View style={styles.navBarBottom}>
          <TouchableOpacity onPress={this.onPressBack} style={styles.button}>
            <View>
              <Text style={styles.buttonText}>Back</Text>
            </View>
          </TouchableOpacity>
          <TouchableOpacity
            onPress={this.onPressSaveTask}
            style={styles.button}
          >
            <View>
              <Text style={styles.buttonText}>Save task</Text>
            </View>
          </TouchableOpacity>
        </View>
      </View>
    );
  }

  onPressBack = () => {
    this.state.parent.handlePressBack();
  };
}

export const stylesEditScreen = StyleSheet.create({
  textInputField: {
    height: 60,
    margin: 10
  },
  pickerField: {
    margin: 30
  }
});
