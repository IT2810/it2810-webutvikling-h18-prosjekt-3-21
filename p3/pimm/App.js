import React from "react";
import { StyleSheet, Text, View } from "react-native";
import { HomeScreen } from "./HomeScreen";
import { EditTaskScreen } from "./EditTaskScreen";

export default class App extends React.Component {
  constructor(props) {
    super(props);
  }
  state = {
    currentTab: "HomeScreen"
  };
  handleAddTask = () => {
    this.setState({ currentTab: "AddTaskScreen" });
  };
  handlePressBack = () => {
    this.setState({ currentTab: "HomeScreen" });
  };
  handleEditTask = (id, desc) => {
    displayedScreen = (
      <EditTaskScreen parent={this} taskid={id} taskdesc={desc} />
    );
    this.setState({ currentTab: "EditTaskScreen" });
  };

  render() {
    if (this.state.currentTab === "HomeScreen") {
      displayedScreen = <HomeScreen parent={this} />;
    } else if (this.state.currentTab === "AddTaskScreen") {
      displayedScreen = <EditTaskScreen parent={this} />;
    }
    return displayedScreen;
  }
}