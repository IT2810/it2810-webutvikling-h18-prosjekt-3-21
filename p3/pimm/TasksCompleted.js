import React, { Component } from "react";
import { ScrollView, AsyncStorage } from "react-native";
import { styles } from "./HomeScreen";
import { Task } from "./Task";

export class TasksCompleted extends Component {
  constructor(props) {
    super(props);
    this.state.parent = props.parent;
  }

  state = {
    tasksArray: [],
    parent: null
  };
  _retrieveData = async () => {
    let taskArr = [];
    try {
      AsyncStorage.getAllKeys((err, keys) => {
        AsyncStorage.multiGet(keys, (err, stores) => {
          stores.map((result, i, store) => {
            let task = JSON.parse(store[i][1]);
            if (task !== null) {
              if (task.isCompleted === true) {
                taskArr.push(task);
              }
            }
          });
        }).then(() => {
          this.setState({
            tasksArray: taskArr
          });
        });
      });
    } catch (error) {
      console.log(error);
      alert("Error retrieving data");
    }
  };

  componentDidMount() {
    this._retrieveData();
  }

  updateTasks() {
    this._retrieveData();
  }

  render() {
    return (
      <ScrollView
        contentContainerStyle={styles.taskContainer}
        scrollEnabled={true}
        alwaysBounceVertical={false}
      >
        {this.state.tasksArray.map(task => (
          <Task
            id={task.id_str}
            taskdescription={task.taskDesc}
            key={task.id}
            parent={this}
          />
        ))}
      </ScrollView>
    );
  }
}
