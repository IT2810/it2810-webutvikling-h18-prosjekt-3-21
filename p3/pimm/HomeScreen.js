import React, { Component } from "react";
import { StyleSheet, Text, View, TouchableOpacity, Image } from "react-native";
import { Task } from "./Task";
import { TasksToDo } from "./TasksToDo";
import { TasksCompleted } from "./TasksCompleted";

export class HomeScreen extends Component {
  constructor(props) {
    super(props);
    this.state.parent = props.parent;
  }

  state = {
    displayedTab: "todo",
    tasksArray: []
  };

  updateTasks() {
    this._retrieveData();
  }

  render() {
    let tasks;
    if (this.state.displayedTab === "todo") {
      tasks = <TasksToDo parent={this} />;
    } else if (this.state.displayedTab === "completed") {
      tasks = <TasksCompleted parent={this} />;
    }

    let addTaskButton;
    if (this.state.displayedTab === "todo") {
      addTaskButton = (
        <TouchableOpacity
          onPress={this.onPressAddTask}
          style={styles.addTaskButton}
        >
          <Image
            source={require("./assets/plus.png")}
            style={styles.addTaskButtonImage}
          />
        </TouchableOpacity>
      );
    }
    return (
      <View style={styles.container}>
        <View style={styles.navBarTodo}>
          <TouchableOpacity
            onPress={this.onPressToDo}
            style={[
              this.state.displayedTab == "todo"
                ? styles.activeButton
                : styles.button,
              { borderLeftWidth: 0 }
            ]}
          >
            <View>
              <Text
                style={
                  this.state.displayedTab == "todo"
                    ? styles.activeButtonText
                    : styles.buttonText
                }
              >
                TO DO
              </Text>
            </View>
          </TouchableOpacity>
          <TouchableOpacity
            onPress={this.onPressCompleted}
            style={[
              this.state.displayedTab == "completed"
                ? styles.activeButton
                : styles.button,
              { borderRightWidth: 0 }
            ]}
          >
            <View>
              <Text
                style={
                  this.state.displayedTab == "completed"
                    ? styles.activeButtonText
                    : styles.buttonText
                }
              >
                COMPLETED
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View style={styles.mainContainer}>{tasks}</View>
        {addTaskButton}
      </View>
    );
  }
  onPressToDo = () => {
    this.setState({ displayedTab: "todo" });
  };
  onPressCompleted = () => {
    this.setState({ displayedTab: "completed" });
  };
  onPressAddTask = () => {
    this.state.parent.handleAddTask();
  };
}
export default HomeScreen;
export const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "lightblue",
    justifyContent: "center",
    alignItems: "center",
    marginTop: 0 //20
  },

  mainContainer: {
    flex: 7,
    width: "100%",
    marginTop: 10,
    marginBottom: 20
  },

  taskContainer: {
    margin: 1,
    alignItems: "center",
    paddingBottom: 100
  },

  navBarTodo: {
    flex: 1,
    flexDirection: "row",
    padding: 0,
    marginTop: 30
  },

  button: {
    flex: 1,
    backgroundColor: "#5176A1",
    borderColor: "#29436d",
    width: "100%",
    borderWidth: 1,
    borderRadius: 0,
    alignItems: "center",
    justifyContent: "center",
    marginTop: 3
  },

  activeButton: {
    flex: 1,
    backgroundColor: "lightblue",
    borderColor: "#29436d",
    width: "100%",
    borderWidth: 1,
    borderBottomWidth: 0,
    alignItems: "center",
    justifyContent: "center"
  },

  buttonText: {
    fontSize: 29,
    // fontWeight: "bold",
    color: "#444",
    textShadowColor: "#29436d",
    textShadowRadius: 0,
    textShadowOffset: { width: 1, height: 1 }
  },

  activeButtonText: {
    fontSize: 30,
    //fontWeight: "200",
    color: "#fff",
    textShadowColor: "#29436d",
    textShadowRadius: 0,
    textShadowOffset: { width: 1, height: 1 }
  },

  addTaskButton: {
    alignItems: "center",
    justifyContent: "center",
    position: "absolute",
    bottom: "5%",
    right: "5%",
    padding: 0
  },

  addTaskButtonImage: {
    width: 60,
    height: 60
  }
});
