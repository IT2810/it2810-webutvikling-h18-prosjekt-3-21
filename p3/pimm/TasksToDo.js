import React, { Component } from "react";
import { ScrollView, AsyncStorage } from "react-native";
import { styles } from "./HomeScreen";
import { Task } from "./Task";

export class TasksToDo extends Component {
  constructor(props) {
    super(props);
    this.state.parent = props.parent;
  }

  state = {
    tasksArray: [],
    parent: null
  };

  _retrieveData = async () => {
    let taskArr = [];
    try {
      AsyncStorage.getAllKeys((err, keys) => {
        AsyncStorage.multiGet(keys, (err, stores) => {
          stores.map((result, i, store) => {
            let task = JSON.parse(store[i][1]);
            if (task.isCompleted === false) {
              taskArr.push(task);
            }
          });
        }).then(() => {
          this.setState({
            tasksArray: taskArr
          });
        });
      });
    } catch (error) {
      console.log(error);
      alert("Error retrieving data");
    }
  };

  componentDidMount() {
    this._retrieveData();
  }

  updateTasks() {
    this._retrieveData();
  }

  render() {
    console.log("TasksToDo: ", this.state.tasksArray);
    return (
      <ScrollView
        contentContainerStyle={styles.taskContainer}
        scrollEnabled={true}
        alwaysBounceVertical={false}
      >
        {this.state.tasksArray.map(task => (
          <Task
            id={task.id}
            taskdescription={task.taskDesc}
            key={task.id}
            parent={this}
          />
        ))}
      </ScrollView>
    );
  }
}
