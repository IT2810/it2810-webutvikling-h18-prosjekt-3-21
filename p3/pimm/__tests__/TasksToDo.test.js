import * as TestRenderer from "react-test-renderer";
import React from "react";
import View from "react-native";
//import TestRenderer from "react-test-renderer";
import App from "../App";
import HomeScreen from "../HomeScreen";
import PedometerSensor from "../PedoSensor";
import EditTaskScreen from "../EditTaskScreen";
import TasksCompleted from "../TasksCompleted";
import TasksToDo from "../TasksToDo";
import Task from "../Task";

beforeAll(() => {
  tree = TestRenderer.create(<TasksToDo />);
  TasksToDoTest = TestRenderer.create(<TasksToDo />).getInstance();
});

test("Methods in TasksToDo.js works correctly", () => {
  // Snapshot test
  expect(tree.toJSON).toMatchSnapshot();
});
