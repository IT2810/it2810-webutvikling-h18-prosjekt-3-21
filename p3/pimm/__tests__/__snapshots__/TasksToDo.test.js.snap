// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Methods in TasksToDo.js works correctly 1`] = `[Function]`;
